crate/cli/src/actions/cover_crypt/policy.rs

```diff
@@ -93,19 +91,39 @@ impl TryFrom<&Policy> for PolicySpecifications {
 
     fn try_from(policy: &Policy) -> Result<Self, Self::Error> {
         let mut result: HashMap<String, Vec<String>> = HashMap::new();
-        for (axis_name, params) in &policy.axes {
-            let axis_full_name =
-                axis_name.clone() + if params.is_hierarchical { "::+" } else { "" };
-            let mut attributes = Vec::with_capacity(params.attribute_names.len());
-            for att in &params.attribute_names {
-                let name = att.clone()
-                    + match policy.attribute_hybridization_hint(&Attribute::new(axis_name, att))? {
-                        EncryptionHint::Hybridized => "::+",
-                        EncryptionHint::Classic => "",
-                    };
-                attributes.push(name);
-            }
-            result.insert(axis_full_name, attributes);
+        for (dim_name, dimension) in &policy.dimensions {
+            let mut dim_full_name = dim_name.clone();
+            let attributes = if let Some(ordered_attrs) = &dimension.order {
+                dim_full_name += "::+";
+                ordered_attrs
+                    .iter()
+                    .map(|name| {
+                        name.clone()
+                            + dimension
+                                .attributes
+                                .get(name)
+                                .map(|attr| match attr.encryption_hint {
+                                    EncryptionHint::Hybridized => "::+",
+                                    EncryptionHint::Classic => "",
+                                })
+                                .unwrap_or("")
+                    })
+                    .collect()
+            } else {
```

Comment:
This is needed only because you need to reorder the attributes in `dimension.attributes`. Therefore it would be made useless if you had used an enum (you could directly iterate on the correct attribute order). Since you don't want to go that way, at least extract the reordering part from the name building logic to make this code simpler:
```rust
// you could write the "get attribute in an ordered fashion
// (or not)" in the CoverCrypt repository.
let attributes = dimension.get_attributes();
// The issue is that if the above function returns a `Vec`, you
// allocate twice and go through attributes twice. A solution
// could be to return an iterator on said attributes. This avoids
// both the double collecting and iteration issues. Maybe you
// could rename `get_attributes()` into `attributes()` to mimic the
// hash map's `.values()` method which also returns an iterator? 
let attributes = attributes.map(|(name, attr)| {
    // the rest goes unchanged
}).collect();
```